analyzer/src/main/kotlin/managers/Unmanaged.kt

```diff
@@ -102,7 +102,7 @@ class Unmanaged(
 
             else -> {
                 // For all non-GitRepo VCSes derive the name from the VCS URL.
-                val fileUrl = File(vcsInfo.url).takeIf { it.exists() } ?: File(URI(vcsInfo.url))
+                val fileUrl = File(URI(vcsInfo.url).path)
```

Comment:
Just for my understanding, could you tell where exactly the mentioned exception got thrown ?

Reply:
It was actually not that easy to debug. The problem manifested itself at the very end of the analyzer run with a stack-trace as below. With the `--stack-trace` option enabled, it can be seen that the construction of the `File` from the `URI` after the Elvis operator threw an `IllegalArgumentException`, as the URI could not be converted to a file.

```
Exception in thread "main" java.lang.IllegalArgumentException: The VcsInfo(type=, url=, revision=, path=) of project 'Unmanaged::.:' cannot be found in Repository(vcs=VcsInfo(type=Git, url=XXXX.
	at org.ossreviewtoolkit.model.OrtResult.getFilePathRelativeToAnalyzerRoot(OrtResult.kt:245)
	at org.ossreviewtoolkit.model.OrtResult.getDefinitionFilePathRelativeToAnalyzerRoot(OrtResult.kt:231)
	at org.ossreviewtoolkit.model.config.Excludes.findPathExcludes(Excludes.kt:52)
	at org.ossreviewtoolkit.model.OrtResult$projects$2.invoke(OrtResult.kt:104)
	at org.ossreviewtoolkit.model.OrtResult$projects$2.invoke(OrtResult.kt:100)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at org.ossreviewtoolkit.model.OrtResult.getProjects(OrtResult.kt:100)
	at org.ossreviewtoolkit.model.OrtResult.getProject(OrtResult.kt:321)
	at org.ossreviewtoolkit.model.OrtResult.isProject(OrtResult.kt:472)
	at org.ossreviewtoolkit.model.OrtResult.isExcluded(OrtResult.kt:269)
	at org.ossreviewtoolkit.model.OrtResult.getProjects(OrtResult.kt:341)
	at org.ossreviewtoolkit.model.OrtResult.getProjects$default(OrtResult.kt:339)
	at org.ossreviewtoolkit.cli.commands.AnalyzerCommand.run(AnalyzerCommand.kt:242)
	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:198)
	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:211)
	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:18)
	at com.github.ajalt.clikt.core.CliktCommand.parse(CliktCommand.kt:400)
	at com.github.ajalt.clikt.core.CliktCommand.parse$default(CliktCommand.kt:397)
	at com.github.ajalt.clikt.core.CliktCommand.main(CliktCommand.kt:415)
	at com.github.ajalt.clikt.core.CliktCommand.main(CliktCommand.kt:440)
	at org.ossreviewtoolkit.cli.OrtMainKt.main(OrtMain.kt:76)
```

Reply:
Given this stack trace, to my understanding, it's not possible that [1] introduced this bug (because it did not affect any VCS info). By coincidence I've been debugging recently as similar stack trace, and I believe this  stacktrace only happens if the VCS info of the project is not contained in the VCSes listed in `ortResult.repository`.

In particular, the function that crashes assumes that both, the project's VCS as well as the VCSes in `ortResult.repository` are normalized, see [2]. Setting a breakpoint in there IMO should help, assuming you have an Ort file to reproduce the issue.

[1] https://github.com/oss-review-toolkit/ort/commit/07c8633ac7a67c0ec8f3cb99da2fd387e3eee4fa 
[2] https://github.com/oss-review-toolkit/ort/blob/fb1d0a91aa89e6afa05b50c62bf311aaedb5fa03/model/src/main/kotlin/Repository.kt#L71-L81

Reply:
Given this stack trace, to my understanding, it's not possible that [1] introduced this bug (because it did not affect any VCS info). By coincidence I've been debugging recently as similar stack trace, and I believe this  stacktrace only happens if the VCS info of the project is not contained in the VCSes listed in `ortResult.repository`.

In particular, the function that crashes assumes that both, the project's VCS as well as the VCSes in `ortResult.repository` are normalized, see [2]. Setting a breakpoint in there IMO should help, assuming you have an Ort file to reproduce the issue.

[1] https://github.com/oss-review-toolkit/ort/commit/07c8633ac7a67c0ec8f3cb99da2fd387e3eee4fa 
[2] https://github.com/oss-review-toolkit/ort/blob/fb1d0a91aa89e6afa05b50c62bf311aaedb5fa03/model/src/main/kotlin/Repository.kt#L71-L81